github-actions/generate-doc/action.yml

```diff
@@ -41,7 +41,5 @@ runs:
         cd $APP_REPO_DIR
         git config user.name "splunk-soar-connectors-admin"
         git config user.email "admin@splunksoar"
-        git add README.md
-        git commit -m "Update README.md"
-        git push
+        git add README.md || git diff --quiet && git diff --staged --quiet || git commit -am 'Update README.md' || git push
```

Comment:
are the `&&` and `||` uses here backwards? also potentially missing parentheses around the inner `git diff` conditions

`A || B || C` -> B runs only if A fails, C runs only if A and B fails
`A && B && C` -> B runs only if A succeeds, C runs only if A and B succeds

Reply:
Updated bash commands and tested various scenarios. Please feel free to comment any new test cases. Thank you!